### PR TITLE
Export Service - add simple manifest file, manifest.txt

### LINKED ISSUE #695: Data Export - manifest.txt
## Summary
Non technical users find it hard to understand `manifest.json` so we want to provide them an alternative `manifest.txt`. This should look similar to `manifest preview` but additionally it should show the details contained in the `manifest.json`.

Example:
image

```json
{
  "type": "Samples Export",
  "date": "2024-08-09",
  "children": [
    {
      "name": "INXT_PRJ_AYIJBKDPRH",
      "type": "folder",
      "irida-next-type": "project",
      "irida-next-name": "Outbreak 2022",
      "children": [
        {
          "name": "INXT_SAM_AYIJBKDTXV",
          "type": "folder",
          "irida-next-type": "sample",
          "irida-next-name": "Bartonella henselae/Outbreak 2022 Sample 10",
          "children": [
            {
              "name": "INXT_ATT_AYIJAYRD6E",
              "type": "folder",
              "irida-next-type": "attachment",
              "children": [
                {
                  "name": "08-5578-small_S1_L001_R1_001.fastq.gz",
                  "type": "file",
                  "metadata": {
                    "format": "fastq",
                    "direction": "forward",
                    "type": "illumina_pe"
                  }
                },
                {
                  "name": "08-5578-small_S1_L001_R2_001.fastq.gz",
                  "type": "file",
                  "metadata": {
                    "format": "fastq",
                    "direction": "reverse",
                    "type": "illumina_pe"
                  }
                }
              ]
            },
            {
              "name": "INXT_ATT_AYIJBKDT57",
              "type": "folder",
              "irida-next-type": "attachment",
              "children": [
                {
                  "name": "reference.fasta",
                  "type": "file",
                  "metadata": {
                    "format": "fasta",
                    "type": "assembly"
                  }
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```

```txt
type: Samples Export
date: 2024-08-09

contents:
b5a78c8b-f5bc-460e-bcdd-d94a3390e0f1/
├─ manifest.json
├─ manifest.txt
├─ INXT_PRJ_AYIJBKDPRH/ (Outbreak 2022)
│  ├─ INXT_SAM_AYIJBKDTXV/ (Bartonella henselae/Outbreak 2022 Sample 10)
│  │  ├─ INXT_ATT_AYIJAYRD6E/
│  │  │  ├─ 08-5578-small_S1_L001_R1_001.fastq.gz
│  │  │  ├─ 08-5578-small_S1_L001_R2_001.fastq.gz
│  │  ├─ INXT_ATT_AYIJBKDT57/
│  │  │  ├─ reference.fasta
```

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## What does this PR do and why?
_Describe in detail what your merge request does and why._
Fixes #695 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._
Example Output of `manifest.txt`

```txt
Samples Export
2024-08-14

contents:
83c2d856-9924-496d-8628-9b97d01a56ee/
├─ INXT_PRJ_AYINCIYSP6/ (Outbreak 2022)
│  ├─ INXT_SAM_AYINCIYSVZ/ (Yersinia pseudotuberculosis/Outbreak 2022 Sample 2)
│  │  ├─ INXT_ATT_AYINBXGCXT/
│  │  │  ├─ 08-5578-small_S1_L001_R1_001.fastq.gz
│  │  │  └─ 08-5578-small_S1_L001_R2_001.fastq.gz
│  │  └─ INXT_ATT_AYINCIYSXQ/
│  │     └─ reference.fasta
│  └─ INXT_SAM_AYINCIYSZA/ (Yersinia pseudotuberculosis/Outbreak 2022 Sample 3)
│     ├─ INXT_ATT_AYINCIYS24/
│     │  └─ reference.fasta
│     └─ INXT_ATT_AYINBXGC26/
│        ├─ 08-5578-small_S1_L001_R2_001.fastq.gz
│        └─ 08-5578-small_S1_L001_R1_001.fastq.gz
└─ INXT_PRJ_AYINCIYTUF/ (Outbreak 2023)
   └─ INXT_SAM_AYINCIYUVE/ (Yersinia pseudotuberculosis/Outbreak 2023 Sample 10)
      ├─ INXT_ATT_AYINBXGEXA/
      │  ├─ 08-5578-small_S1_L001_R1_001.fastq.gz
      │  └─ 08-5578-small_S1_L001_R2_001.fastq.gz
      └─ INXT_ATT_AYINCIYUW6/
         └─ reference.fasta
```

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Go to a project namespace, select Samples from the sidebar
2. Check some boxes and create a Sample Export
3. After if completes, download the zip file an inspect the `manifest.txt` inside.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good!

File: test/jobs/data_exports/create_job_test.rb [hunk 1]
```diff
@@ -82,6 +83,46 @@ def setup
...
+          assert expected_files_in_zip.include?(entry.to_s)
+          actual_simple_manifest = entry.get_input_stream.read.force_encoding('utf-8') if entry.to_s == 'manifest.txt'
+          expected_files_in_zip.delete(entry.to_s)
+        end
+      end
+      assert expected_files_in_zip.empty?
+
+      expected_simple_manifest = [
+        'Samples Export',
+        '2024-08-14',
```
Comment (ChrisHuynh333) [inline test/jobs/data_exports/create_job_test.rb:110]: Didn't notice until a failed test just now, this should be like other manifest tests `Date.current.strftime('%Y-%m-%d')`

**Review by ChrisHuynh333 — CHANGES_REQUESTED** [review]

File: test/jobs/data_exports/create_job_test.rb (hunk 1 above)
Comment (JeffreyThiessen) [reply test/jobs/data_exports/create_job_test.rb]: Fixed in https://github.com/phac-nml/irida-next/pull/700/commits/85c522dfbbcf65cd1f84bddf669f0413d6a990e5

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good!

**Pull request merged**